omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/UserResource2_0Test.java

```diff
@@ -195,4 +197,29 @@ public void testSearchingByUserAndRole() {
 		assertSearch("doesNotExist", Arrays.asList("System Developer"), 0);
 	}
 	
+	@Test
+	public void testChangeUsersOwnPassword() throws Exception {
```

Comment:
If we need to authenticate with the user, we need to have edit users, get locations, get users privileges for saveUser method. We tried to find the user with those privileges. Even when we assign those privileges it is refreshing the authenticated user.